File: src/lib/api/cuentaCorriente.ts
import { invoke } from "./client";

export type MovimientoCuentaCorriente = {
  id: number;
  tipo: "venta_fiada" | "pago" | "ajuste" | "devolucion";
  monto: number;
  saldoResultante: number;
  metodoPagoNombre: string | null;
  fecha: string;
  observacion: string | null;
};

export function listarMovimientos(clienteId: number) {
  return invoke<MovimientoCuentaCorriente[]>("cuenta_corriente_listar_movimientos", { clienteId });
}

export function registrarPago(
  clienteId: number,
  monto: number,
  metodoPagoId: number,
  observacion: string | null,
) {
  return invoke<void>("cuenta_corriente_registrar_pago", {
    clienteId,
    datos: { monto, metodoPagoId, observacion },
  });
}

export function ajustarCuentaCorriente(clienteId: number, monto: number, motivo: string) {
  return invoke<void>("cuenta_corriente_ajustar", {
    clienteId,
    datos: { monto, motivo },
  });
}

File: src-tauri/src/models/cuenta_corriente.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Serialize, sqlx::FromRow)]
#[serde(rename_all = "camelCase")]
pub struct MovimientoCuentaCorriente {
    pub id: i64,
    /// venta_fiada | pago | ajuste | devolucion
    pub tipo: String,
    pub monto: i64,
    pub saldo_resultante: i64,
    pub metodo_pago_nombre: Option<String>,
    pub fecha: String,
    pub observacion: Option<String>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RegistrarPago {
    /// Centavos, siempre positivo -- lo que pagó el cliente.
    pub monto: i64,
    pub metodo_pago_id: i64,
    pub observacion: Option<String>,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct AjusteCuentaCorriente {
    /// Centavos. Positivo aumenta la deuda (ej. migrar saldo previo desde
    /// papel), negativo la reduce (ej. corrección, condonación).
    pub monto: i64,
    pub motivo: String,
}
